dev/com.ibm.ws.microprofile.openapi.ui/src/com/ibm/ws/microprofile/openapi/ui/OpenAPIUIEndpointManager.java

```diff
@@ -0,0 +1,192 @@
+        }
+        openAPIWABConfigManager = new OpenAPIWABConfigManager(context, OPEN_API_UI_VAR_Name, path);
+        openAPIWABConfigManager.pushConfiguration();
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context, int reason){
+        openAPIWABConfigManager.deactivate();
+    }
+
+    protected String processOpenAPIUIPATH(final ComponentContext context, final Config config) {
+        String uiPath = getOptionalValue(config, OPEN_API_UI_CONFIG_NAME, String.class, OPEN_API_UI_PATH);
+
+        uiPath = resolvePath(uiPath);
+
+        // Check if there are potential conflicts with other Liberty/MicroProfile features and warn if there is a match.
+        if (CONFLICTING_PATHS_MAP.keySet().contains(uiPath)) {
+            //just warn that might clash with other features as they might not be active, so not a cause for failure
+            Tr.warning(tc, "CWWKO1752W_OPEN_API_UI_PATH_POTENTIAL_CONFLICT", CONFLICTING_PATHS_MAP.get(uiPath));
+        }
+
+        //Validate URL for any paths that we should revert to default to
+        if(!validatePath(config, uiPath)){
+            uiPath = OPEN_API_UI_PATH;
+        }
+
+        return uiPath;
+    }
+
+    protected boolean validatePath(Config config, String path){
+        boolean valid=true;
+        //check against OpenAPI Data endpoint
+        String dataEndpoint = getOptionalValue(config, OPEN_API_ENDPOINT_CONFIG_NAME, String.class, OPEN_API_ENDPOINT_RESERVED_PATH);
+        if (path.equals(dataEndpoint)) {
+            Tr.warning(tc, "CWWKO1750W_OPEN_API_UI_PATH_RESERVED", OPEN_API_ENDPOINT_RESERVED_PATH);
+            valid=false;
+        }
+        //check against known invalid paths that contain valid characters
+        else if (Arrays.asList(INVALID_PATHS).contains(path)){
+            Tr.warning(tc, "CWWKO17510W_OPEN_API_UI_PATH_INVALID", path);
+            valid=false;
+        }
+        //check that path is made up of valid characters
+        else if(!PATH_PATTERN.matcher(path).matches()){
+            Tr.warning(tc, "CWWKO1751W_OPEN_API_UI_PATH_INVALID",path);
+            valid=false;
+        //check against other feature paths to see if a potential conflict might occur
+        }
+        return valid;
+    }
+
+    protected String resolvePath(String path){
+        // Add a forward slash if the path does not already start with one.
+        if (!path.startsWith("/")) {
+            path = '/' + path;
+        }
+        //in case someone specifies for example`////foo//bar` which will get through, keep running until all potential double `//` are accounted for
+        // given first pass will go to //foo/bar
+        while(path.contains("//")){
+            path = path.replace("//","/");
+        }
+        // Remove trailing slash if the path contains one.
+        if (path.endsWith("/") && !path.equals("/")) {
+            path = path.substring(0, path.length() - 1);
+        }
+        return path;
+    }
```

Comment:
Should these three "processing" methods be static and have separate unit tests to check all the parsing edge cases and invalid path values?

Reply:
`validatePath` and `resolvePath` certainly should have unit tests. Though both were going to end up in common code when I did a bit of refactoring for the data endpoint as both have largely the same requirements
`validatePath` would need a small refactor to make it not reliant on config existing and that would allow it to also be `static`

The initial processing potentially could do, but would say moving the warning for potential conflicts out so that was more testable would be a better way to go.